### DISCUSSION TITLE
Stokes flow pressure display problems

### DISCUSSION THREAD

**Discussion by likeduck** [PMEAL/OpenPNM#2753] [Q&A]
Hi prof，
  Is there anyway to diaplay colorbar in the figure of Stokes flow's pressure? I got independent one like the following image：
![colorbar](https://github.com/PMEAL/OpenPNM/assets/117707506/305149e6-ad0e-4dd5-b328-925f77d5dff2)
I try many times in 3D PNM but failed. Could you give me some idea?

Thank you so much!

likeduck

**Comment by jgostick** [comment]
This seems like a matplotlib problem no?  I was actually learning how to use colorbars recently, like this:

```
fig, ax = plt.subplots()
handle = ax.imshow(somthing here)
plt.colorbar(handle)
```

The point is you have to catch the 'handle' returned when calling imshow, so that you can pass it to the colorbar function.

**Reply by likeduck** [reply]
OOOh! It is indeed the problem of matplotlib. I am learning it, too. And I find that there was 1 dimension in `phase['pore.pressure']` so the `ax.imshow()` didn't work. When I extend `phase['pore.pressure']` to 2dimension by copying itself, the `ax.imshow()` failed to plot. Anyway, it was an attractive question！I will learn more deeply.
Thanks for your kind reply!
